src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -320,17 +332,25 @@ private void EventLoop()
                         {
                             Debug.Assert(handle.ToInt64() < MaxHandles.ToInt64(), $"Unexpected values: handle={handle}, MaxHandles={MaxHandles}");
 
-                            Interop.Sys.SocketEvents events = context.HandleSyncEventsSpeculatively(socketEvent.Events);
-                            if (events != Interop.Sys.SocketEvents.None)
+                            if (context.PreferInlineCompletions)
+                            {
+                                context.HandleEventsInline(socketEvent.Events);
+                            }
+                            else
```

Comment:
Presumably this branch is well-predicted and thus doesn't negatively impact throughput in the common case where PreferInlineCompletions is false?  If it does, we may need to duplicate the foreach loop, such that the branch moves to outside of the foreach.

Before we merge, can we do a run of the benchmarks to confirm there's no regression introduced by this?